dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -79,6 +79,7 @@ def show
     @current_user = current_user && User.includes(:teachers).where(id: current_user.id).first
     authorize! :read, ScriptLevel
     @script = ScriptLevelsController.get_script(request)
+    @tts_autoplay_enabled = current_user&.sections_as_student&.map(&:tts_autoplay_enabled)&.reduce(false, :|)
```

Comment:
Could you elaborate a bit on why @tts_autoplay_enabled is determined this way?  Why do we want @tts_autoplay_enabled to be true if any section that a student is in has enabled tts?

Reply:
This was inherited code, but what would be more accurate here might be to check if the ScriptLevel is part the work assigned through the section. There are probably some edge cases, like what the student is in 2 different sections and one has autoplay enabled and one doesn't, should it autoplay? I guess another thing to consider is, why not autoplay all instructions if a student is enrolled in a section where autoplay is set.

Reply:
Chat with Amanda:

When a user is on a level how should we determine if the instructions should be autoplayed? Currently, if the user is enrolled in any section with autoplay enabled the instructions will be played. Does that sound right to you?

That sounds right, with the added caveat that if the section is archived, we should ignore its settings. This has its problems / complexities, but this is how we handle the 'the student is in multiple sections' issue elsewhere already

Reply:
Updating to exclude archived sections..

Reply:
Ok, I'm curious to learn more about why we decided to do it this way, but we'll save that conversation for another day.